src/inmanta/data/__init__.py

```diff
@@ -4725,6 +4725,37 @@ async def set_deployed_multi(
         async with cls.get_connection(connection) as connection:
             await connection.execute(query, environment, version, resource_ids)
 
+    @classmethod
+    async def reset_resource_state(
+        cls,
+        environment: uuid.UUID,
+        version: int,
+        *,
+        connection: Optional[asyncpg.connection.Connection] = None,
+    ) -> None:
+        """
+        Update resources on the latest version of the model stuck in "deploying" state. The status will be reset to the latest
+        non deploying status.
+
+        :param environment: The environment impacted by this
+        :param version: The version of the model impacted by this
+        :param connection: The connection to use
+        """
+        query = """
+            UPDATE resource AS updated_r
+            SET status=inconsistent_r.last_non_deploying_status::TEXT::resourcestate
+            FROM (
+                SELECT r.resource_id, r.status, r.model, ps.last_non_deploying_status
+                FROM resource r
```

Comment:
I think we should use `{cls.table_name()}` and `{ResourcePersistentState.table_name()}`.